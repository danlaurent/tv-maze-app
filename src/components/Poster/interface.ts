import { StyleProp, ViewStyle } from 'react-native';

export interface IPoster {
  source?: string;
  size?: 'small' | 'large';
  style?: StyleProp<ViewStyle>;
  testID?: string;
}
