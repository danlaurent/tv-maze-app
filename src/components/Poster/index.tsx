import React from 'react';
import { Image, View, StyleSheet, Text } from 'react-native';
import { IPoster } from './interface';

const styles = StyleSheet.create({
  small: {
    width: 105,
    height: 147,
    borderRadius: 4,
  },
  large: {
    width: 210,
    height: 295,
    borderRadius: 4,
  },
  posterPlaceholder: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: '#CECECE',
    justifyContent: 'center',
    alignItems: 'center',
  },
  posterPlaceholderText: {
    fontSize: 24,
    fontWeight: '700',
    textAlign: 'center',
  },
});

const Poster = ({ source, size = 'small', style }: IPoster) => (
  <View style={StyleSheet.flatten([styles[size], style])}>
    {source ? (
      <Image source={{ uri: source }} style={styles[size]} />
    ) : (
      <View style={styles.posterPlaceholder}>
        <Text style={styles.posterPlaceholderText}>N/A</Text>
      </View>
    )}
  </View>
);

export default Poster;
