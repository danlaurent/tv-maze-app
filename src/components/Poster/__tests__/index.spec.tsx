import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import Poster from '..';

describe('Poster', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <Poster source='https://via.placeholder.com/300' testID='testPoster' />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When source is not provided', () => {
    beforeEach(() => {
      tree = render(<Poster />);
    });
    it('shows an image placeholder', () => {
      expect(tree.queryByTestId('testPoster_imagePlaceholder')).toBeNull();
    });
  });
});
