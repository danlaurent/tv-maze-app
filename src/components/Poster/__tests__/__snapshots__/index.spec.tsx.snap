// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 4,
      "height": 147,
      "width": 105,
    }
  }
  testID="testPoster"
>
  <Image
    source={
      Object {
        "uri": "https://via.placeholder.com/300",
      }
    }
    style={
      Object {
        "borderRadius": 4,
        "height": 147,
        "width": 105,
      }
    }
    testID="testPoster_image"
  />
</View>
`;
